File: 1046-last-stone-weight/1046-last-stone-weight.py
class Solution:
    def lastStoneWeight(self, stones):
        modified = [stone * -1 for stone in stones]

        heapq.heapify(modified)

        while len(modified) > 1:
            biggest = heapq.heappop(modified) * -1
            secBiggest = heapq.heappop(modified) * -1
            
            if biggest != secBiggest:
                toInsert = (biggest - secBiggest) * -1
                heapq.heappush(modified,toInsert)
        res = [num * -1 for num in modified]
        if res:
            return res[0]
        else:
            return 0

File: 1046-last-stone-weight/1046-last-stone-weight.cpp
class Solution {
public:
    int lastStoneWeight(vector<int>& stones) {
        priority_queue<int> pq(stones.begin(), stones.end());
        int s1, s2;
        while (pq.size() > 1) {
            s1 = pq.top();
            pq.pop();
            s2 = pq.top();
            pq.pop();

            if (s1 != s2) {
                if (s1 > s2)  pq.push(s1 - s2);
                else pq.push(s2 - s1);
            }
        }
        if (pq.empty()) return 0;
        else return pq.top();
    }
};